numpyro/distributions/constraints.py

```diff
@@ -243,8 +280,16 @@ def __repr__(self):
     def feasible_like(self, prototype):
         return jax.numpy.broadcast_to(self.lower_bound + 1, jax.numpy.shape(prototype))
 
+    def tree_flatten(self):
+        return (self.lower_bound,), (("lower_bound",), dict())
 
-class _Positive(_GreaterThan, _SingletonConstraint):
+    def __eq__(self, other):
+        return isinstance(other, _GreaterThan) & jnp.array_equal(
+            self.lower_bound, other.lower_bound
+        )
+
+
+class _Positive(_SingletonConstraint, _GreaterThan):
     def __init__(self):
         super().__init__(0.0)
```

Comment:
I'm not sure. I feel that this call will not set lower_bound to 0 after you switch the order of parent class. What's the motivation for that? I guess it causes the lint issue.

Reply:
I think it would, since `_SingletonConstraints` just grabs the `positive` object from the global namespace during `__new__` (and my [tests](https://github.com/pierreglaser/numpyro/blob/8ba77674f7a07d479c8c98290e46cb1f4f5334ab/test/test_constraints.py#L23) should test for this). The tests were passing even with that change in, what caused the doc failing was the last 2 commits. That being said, it is true that the error mention `lower_bound`...